### PR TITLE
OpenID Connect support

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
OIDC Support for token management and security

**Which issue(s) this PR fixes** (optional)
Part of #448

**Special notes for your reviewer**:
* Role in claims is now Roles
* New switches added to osd program to make it easier to use
* Reorganized token management
* Abstracted token management in the context to be consumed by Ownership
* Added `username` support to be consumed by Ownership

### CODE REVIEW

File: cmd/osd-token-generator/main.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		return
+	}
+	if len(*config) == 0 {
+		fmt.Println("Must provide a file name")
+		os.Exit(1)
+	}
+
+	// Get claims
+	claims := &auth.Claims{
+		Issuer: *issuer,
```
Comment (adityadani) [inline cmd/osd-token-generator/main.go]: Error out here if the issuer has https://

Comment (adityadani) [inline cmd/osd-token-generator/main.go]: Check if issuer has https:// and error out

File: pkg/auth/oidc.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	}
+	for _, requiredClaim := range requiredClaims {
+		if _, ok := claims[requiredClaim]; !ok {
+			// Claim missing
+			return nil, fmt.Errorf("Required claim %v missing from token", requiredClaim)
+		}
+	}
+
+	// Return claims
+	var sdkClaims Claims
```
Comment (adityadani) [inline pkg/auth/oidc.go]: why not just return claims from line 76

File: pkg/auth/token.go [hunk 3]
```diff
@@ -16,73 +16,60 @@ limitations under the License.
...
-	Issuer string
 }
 
-// Token returns a signed JWT containing the claims provided
-func Token(
-	claims *Claims,
-	signature *Signature,
-	options *Options,
-) (string, error) {
+// TokenType returns the type of token. Values are: os, oidc
```
Comment (adityadani) [inline pkg/auth/token.go]: nit: comment should be for TokenIssuer

File: pkg/auth/oidc.go (hunk 2 above)
Comment (lpabon) [reply pkg/auth/oidc.go]: Because line 76 has no structure, it only gets the first level of the keys.

**Comment by lpabon** [comment]
I'll fix these in the ownership PR which is coming tomorrow.

**Pull request merged**